### ISSUE TITLE
Not 4.7. compatible: Setting "Modus: Merge in Household not configuable"

### ISSUE THREAD

**Issue by bkaless** [systopia/de.systopia.householdmerge#7]
I want to configure the modul to be able to merge contacts in a household and I get the following error:

(CiviCRM 4.7.22 on drupal 7)

`backTrace

#0 /var/www/html/sites/all/modules/civicrm/CRM/Core/Error.php(374): CRM_Core_Error::backtrace()
#1 /var/www/html/sites/all/modules/civicrm/CRM/Admin/Form/Setting.php(225): CRM_Core_Error::fatal("Unrecognized setting. This may be a config field which has not been properly ...")
#2 /var/www/html/sites/all/modules/civicrm/CRM/Admin/Form/Setting.php(172): CRM_Admin_Form_Setting->commonProcess((Array:4))
#3 /var/www/html/sites/default/files/civicrm/ext/de.systopia.householdmerge/CRM/Admin/Form/Setting/Household.php(76): CRM_Admin_Form_Setting->postProcess()
#4 /var/www/html/sites/all/modules/civicrm/CRM/Core/Form.php(447): CRM_Admin_Form_Setting_Household->postProcess()
#5 /var/www/html/sites/all/modules/civicrm/CRM/Core/StateMachine.php(160): CRM_Core_Form->mainProcess()
#6 /var/www/html/sites/all/modules/civicrm/CRM/Core/QuickForm/Action/Next.php(61): CRM_Core_StateMachine->perform(Object(CRM_Admin_Form_Setting_Household), "next", "Next")
#7 /var/www/html/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php(203): CRM_Core_QuickForm_Action_Next->perform(Object(CRM_Admin_Form_Setting_Household), "next")
#8 /var/www/html/sites/all/modules/civicrm/packages/HTML/QuickForm/Page.php(103): HTML_QuickForm_Controller->handle(Object(CRM_Admin_Form_Setting_Household), "next")
#9 /var/www/html/sites/all/modules/civicrm/CRM/Core/Controller.php(351): HTML_QuickForm_Page->handle("next")
#10 /var/www/html/sites/all/modules/civicrm/CRM/Utils/Wrapper.php(113): CRM_Core_Controller->run()
#11 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(284): CRM_Utils_Wrapper->run("CRM_Admin_Form_Setting_Household", "Household Settings", NULL)
#12 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(84): CRM_Core_Invoke::runItem((Array:13))
#13 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(52): CRM_Core_Invoke::_invoke((Array:4))
#14 /var/www/html/sites/all/modules/civicrm/drupal/civicrm.module(448): CRM_Core_Invoke::invoke((Array:4))
#15 [internal function](): civicrm_invoke("admin", "setting", "household")
#16 /var/www/html/includes/menu.inc(527): call_user_func_array("civicrm_invoke", (Array:3))
#17 /var/www/html/index.php(21): menu_execute_active_handler()
#18 {main}

Entschuldigung, eine Bereitstellung ist uns gerade nicht möglich.
Unrecognized setting. This may be a config field which has not been properly migrated to a setting. (hh_mode, hh_head_mode, hh_member_relation, hh_head_relation)

Zur Anfangsseite zurück.
`

**Comment by bkaless** [comment]
Perhaps householdtools are not 4.7. ready?

**Comment by bjendres** [comment]
No, probably isn't, sorry. But is says ``4.6`` only in the ``info.xml``...

**Comment by bkaless** [comment]
I will test my issue in a 4.6 installation, after that we know if the error is 4.7 related.

**Comment by bjendres** [comment]
Thanks @bkaless

**Comment by bkaless** [comment]
Yes, it is related to 4.7. In 4.6 the extension is configurable, in 4.7 it isn't.

**Comment by bjendres** [comment]
Yes, that's what I suspected. They changed the way the settings work between 4.6 and 4.7. Let's see when we can find time and funds to fix it.

**Comment by bkaless** [comment]
Could you estimate the costs to change that?

**Comment by bjendres** [comment]
I can't estimate well, since I'm not quite sure what exactly is the problem, but should hopefully be between 1 and 3 hours.

**Comment by bkaless** [comment]
We are able to fund these hours. So please have a look at it. Do you need any formal order from me?

**Comment by bkaless** [comment]
Backtrace on 4.7.29

backTrace
```
#0 /var/www/html/sites/all/modules/civicrm/CRM/Core/Error.php(374): CRM_Core_Error::backtrace()
#1 /var/www/html/sites/all/modules/civicrm/CRM/Admin/Form/Setting.php(224): CRM_Core_Error::fatal("Unrecognized setting. This may be a config field which has not been properly ...")
#2 /var/www/html/sites/all/modules/civicrm/CRM/Admin/Form/Setting.php(171): CRM_Admin_Form_Setting->commonProcess((Array:4))
#3 /var/www/html/sites/default/files/civicrm/ext/de.systopia.householdmerge/CRM/Admin/Form/Setting/Household.php(76): CRM_Admin_Form_Setting->postProcess()
#4 /var/www/html/sites/all/modules/civicrm/CRM/Core/Form.php(447): CRM_Admin_Form_Setting_Household->postProcess()
#5 /var/www/html/sites/all/modules/civicrm/CRM/Core/StateMachine.php(160): CRM_Core_Form->mainProcess()
#6 /var/www/html/sites/all/modules/civicrm/CRM/Core/QuickForm/Action/Next.php(61): CRM_Core_StateMachine->perform(Object(CRM_Admin_Form_Setting_Household), "next", "Next")
#7 /var/www/html/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php(203): CRM_Core_QuickForm_Action_Next->perform(Object(CRM_Admin_Form_Setting_Household), "next")
#8 /var/www/html/sites/all/modules/civicrm/packages/HTML/QuickForm/Page.php(103): HTML_QuickForm_Controller->handle(Object(CRM_Admin_Form_Setting_Household), "next")
#9 /var/www/html/sites/all/modules/civicrm/CRM/Core/Controller.php(351): HTML_QuickForm_Page->handle("next")
#10 /var/www/html/sites/all/modules/civicrm/CRM/Utils/Wrapper.php(113): CRM_Core_Controller->run()
#11 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(283): CRM_Utils_Wrapper->run("CRM_Admin_Form_Setting_Household", "Household Settings", NULL)
#12 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(84): CRM_Core_Invoke::runItem((Array:15))
#13 /var/www/html/sites/all/modules/civicrm/CRM/Core/Invoke.php(52): CRM_Core_Invoke::_invoke((Array:4))
#14 /var/www/html/sites/all/modules/civicrm/drupal/civicrm.module(448): CRM_Core_Invoke::invoke((Array:4))
#15 [internal function](): civicrm_invoke("admin", "setting", "household")
#16 /var/www/html/includes/menu.inc(527): call_user_func_array("civicrm_invoke", (Array:3))
#17 /var/www/html/index.php(21): menu_execute_active_handler()
#18 {main}

Sorry, due to an error, we are unable to fulfill your request at the moment. You may want to contact your administrator or service provider with more details about what action you were performing when this occurred.
Unrecognized setting. This may be a config field which has not been properly migrated to a setting. (hh_mode, hh_head_mode, hh_member_relation, hh_head_relation)
```

**Comment by bjendres** [comment]
This should be fixed with ``1.2.beta1``
